### ISSUE TITLE
Builder init method generated from @Encoding doesn't have @JsonProperty

### ISSUE THREAD

**Issue by chrisbouchard** [immutables/immutables#479]
I have an `@Encoding.Builder` class with an init method, and a `@Value.Immutable` class annotated with `@JsonDeserialize(builder)`. I would like to have the generated builder method annotated with `@JsonProperty("property")` so I can use Jackson builder-based deserialization. I tried annotating the init method itself with `@JsonProperty`, but the annotation isn't copied (and it would need to have its `value` property set, anyway).

Could there be an annotation or annotation property to let Immutables know which builder method to treat as a JSON property? I know encodings aren't really public yet, so I apologize if this is out-of-scope right now.

**Comment by elucash** [comment]
Thank you for trying out experimental functionality and filing the feature request! The general annotation-copy functionality was planned, but not yet implemented. On the other hand, it can be integrated with how we automatically mark builder initializers (non-encoding ones) with `@JsonProperty("property")` when using `@JsonDeserialize(builder = `. In this case, encoding initializer annotated with `@Encoding.Copy` should receive `@JsonProperty("name")` generated and it could be easier to implement than generic annotation copy logic:

```java
@Encoding.Naming(standard = StandardNaming.PUT_ALL)
@Encoding.Init
@Encoding.Copy  // <-- Marks that this is canonical copy initializer
void putAll(Table<R, C, V> table) {
  builder.putAll(table);
}
```

**Comment by chrisbouchard** [comment]
Hooking onto `@Encoding.Copy` would definitely satisfy my use case.

One thing I notice with the built-in `Map` encoding is that the simple init method (`StandardNaming.INIT`) is the one marked with `@JsonProperty`. However, in the example encodings I've seen, including the example in @elucash's comment above, the `PUT_ALL` method is marked with `@Encoding.Copy`. Thus, in this scheme the `PUT_ALL` method would get `@JsonProperty`. Since most of the time one delegates to the other, that's probably fine, but I wonder if it would ever be desireable to have `@JsonProperty` on something other than the `@Encoding.Copy` method.

**Comment by elucash** [comment]
As I see it, there should be not more that one canonical init-copy method. A naming pattern should be irrelevant here, it can be "init" or "put_all", the only limitation is that it should take an argument of the implementation type or its supertype. In this way `@JsonProperty` should be just put on one such method.

**Comment by chrisbouchard** [comment]
That makes sense to me.

**Comment by elucash** [comment]
released in 2.3.9
